src/reducers/index.js

```diff
@@ -1,20 +1,29 @@
-import { INCREMENT, DECREMENT } from '../actions';
+import { INCREMENT, DECREMENT } from "../actions";
 
 const initialState = {
   count: 0
-}
+};
 
 // Our reducer that handles our two action cases:
 // increment and decrement. It receives the state
 // of our redux store, along with an action created
 // by our action creator. What does the reducer
 // need to do with the count in each case?
+
 export default (state = initialState, action) => {
   switch (action.type) {
     case INCREMENT:
-    // Fill in the body of this case
+      return {
+        ...state,
+        count: state.count + 1
+      };
+      break;
     case DECREMENT:
-    // Fill in the body of this case
+      return {
+        ...state,
+        count: state.count - 1
+      };
+      break;
```

Comment:
Perfect! 

So yesterday when we were talking you noticed `++state.count` and you said tried `state.count++`, but got some weird functionality. The reason why this happens is because of the order in which JavaScripts reads.

_e.g_
```
// Increment
let a = 1;
console.log(a++);    // 1
console.log(a);      // 2

// Decrement
let b = 1;
console.log(b--);    // 1
console.log(b);      // 0
```
When you use the increment/decrement operator after the operand, the value will be returned before the operand is increased/decreased.

```
// Increment
let a = 1;
console.log(++a);    // 2
console.log(a);      // 2

// Decrement
let b = 1;
console.log(--b);    // 0
console.log(b);      // 0
```
If you’d rather make the variable increment/decrement before returning, you simply have to use the increment/decrement operator before the operand like above

So in conclusion both ways work and they both update the value, but if the operand comes after the value only the value will be returned unless called on again.

Reply:
Awesome explanation, thanks!